packages/core/src/Input/Input.js

```diff
@@ -38,6 +40,63 @@ const Input = styled.input`
   ${applyVariations('Input')}
 `
 
+const INPUT_ERROR_TEXT = 'InputErrorText'
+const INPUT_ALERT_TEXT = 'InputAlertText'
+const INPUT_INFO_TEXT = 'InputInfoText'
+
+export const Input = (props) => {
+  const { helperText, color, ...restProps } = props
+
+  const getBorderColor = (helperText) => {
+    if (!helperText) {
+      return color
+    }
+    switch (helperText.type.displayName) {
+      case INPUT_ERROR_TEXT:
+        return 'error.base'
+      case INPUT_ALERT_TEXT:
+        return 'alert.base'
+      case INPUT_INFO_TEXT:
+        return 'text.light'
+    }
+  }
```

Comment:
Sorry for playing you about so much with this but I feel like the HelperText should inherit the color from the Input which an option to override if the dev really needs to.

Something like:
`<Input color='error' helperText={<Input.HelperText color='error.dark' />}  />`

And the HelperText could just preconfigure fontSize, weight, margins and other necessary things but all overridable